src/client/response.rs

```diff
@@ -0,0 +1,42 @@
+use crate::msg::Message;
+use anyhow::{Context, Result};
+use async_stream::stream;
+use bb8_redis::{
+    bb8::{Pool, PooledConnection},
+    redis::AsyncCommands,
+    RedisConnectionManager,
+};
+use futures::Stream;
+
+pub struct Response {
+    pool: Pool<RedisConnectionManager>,
+}
+
+impl Response {
+    pub fn new(pool: Pool<RedisConnectionManager>) -> Self {
+        Self { pool }
+    }
+
+    async fn get_connection(&self) -> Result<PooledConnection<'_, RedisConnectionManager>> {
+        let conn = self
+            .pool
+            .get()
+            .await
+            .context("unable to retrieve a redis connection from the pool")?;
+
+        Ok(conn)
+    }
+
+    /// Get response from the remote/local twin
+    /// [call this after use the send function]
+    pub fn response<C: AsRef<str>>(self, ret: C) -> impl Stream<Item = Result<Message>> {
+        stream! {
+            let mut conn = self.get_connection().await.unwrap();
+            let msg: Result<Message> = conn
+            .lpop(ret.as_ref(), None)
```

Comment:
`lpop` is none blocking. it means if there is nothing in the queue this will return None immediately. Which means this loop will basically keep `poping` forever with no mean to stop it I assume this will eat 100% of your cpu. You need to use `brpop` instead. 

We also (to be consistent) always push from the left (lpush) and pop from the right (rpop and brpop)

This way you can easily calculate how long the brpop need to wait based on time out calculations